cdap-common/src/main/java/co/cask/cdap/common/metadata/MetadataRecordV2.java

```diff
@@ -65,26 +67,42 @@ public MetadataRecordV2(NamespacedEntityId entityId, MetadataScope scope, Map<St
     this(entityId.toMetadataEntity(), scope, properties, tags);
   }
 
+  /**
+   * Returns an empty {@link MetadataRecordV2} in the specified {@link MetadataScope} containing the specified
+   * properties and tags
+   */
   public MetadataRecordV2(MetadataEntity metadataEntity, MetadataScope scope, Map<String, String> properties,
                           Set<String> tags) {
-    this.metadataEntity = metadataEntity;
+    this.metadataEntity = new MetadataEntity(metadataEntity);
     this.scope = scope;
-    this.properties = properties;
-    this.tags = tags;
+    this.properties = new HashMap<>(properties);
```

Comment:
I guess now that MetadataEntity is immutable, defensive copy is not needed any more.

Reply:
Done.